src/exports.cc

```diff
@@ -115,16 +147,25 @@ Word get_status(Word code_ptr, Word value_ptr_ptr, Word value_size_ptr) {
 // Continue/Reply/Route
 Word continue_request() {
   auto *context = contextOrEffectiveContext();
+  if (context == nullptr) {
+    return WasmResult::InternalFailure;
+  }
   return context->continueStream(WasmStreamType::Request);
 }
 
 Word continue_response() {
   auto *context = contextOrEffectiveContext();
+  if (context == nullptr) {
+    return WasmResult::InternalFailure;
+  }
```

Comment:
I don't believe that the existing flow can result in context being `nullptr`, so I suspect that you've added those guards here because you've ran into some crashes. Is the context-less execution some quirk / bug introduced in DynVM?

Also, all those changes are technically breaking the existing ABI, since you're introducing new return values to those hostcalls.

Reply:
This is possible from a native module spawning a new thread where the thread local contexts aren't set.